packages/model-viewer/src/three-components/gltf-instance/correlated-scene-graph.ts

```diff
@@ -277,10 +283,14 @@ export class CorrelatedSceneGraph {
          gltfMaterialIndex: number) => {
           updatedMaterials.add(gltfMaterialIndex);
           const gltfElement = this.gltf.materials![gltfMaterialIndex];
-          let threeObjects = this.gltfElementMap.get(gltfElement);
+          if (this[$gltfVariantMap] === null) {
+            this[$gltfVariantMap] = new Map<GLTFElement, ThreeObjectSet>();
+          }
+          let threeObjects = this[$gltfVariantMap]!.get(gltfElement);
           if (threeObjects == null) {
             threeObjects = new Set();
-            this.gltfElementMap.set(gltfElement, threeObjects);
+            // Maps the original gltfElement to a set of possible variants.
+            this[$gltfVariantMap]!.set(gltfElement, threeObjects);
```

Comment:
correlation of element to objects for variants tracked separately, to prevent polluting the `gltfElementMap` when switching variants.

Reply:
Okay, so we're populating the `$gltfVariantMap` but not using it anywhere yet? What is its intended purpose, or is it just keeping things from getting garbage-collected?

Reply:
Wait, I just found where it's used; scratch that.